tasks/build.py

```diff
@@ -67,96 +69,106 @@
 JOB_SOFTWARE_SUBDIR = "software_subdir"
 JOB_OS_TYPE = "os_type"
 
-Job = namedtuple('Job', ('working_dir', 'arch_target', 'repo_id', 'slurm_opts'))
+Job = namedtuple('Job', ('working_dir', 'arch_target', 'repo_id', 'slurm_opts', 'year_month', 'pr_id'))
 
 # global repo_cfg
 repo_cfg = {}
 
 
-def get_build_env_cfg():
+def get_build_env_cfg(cfg):
     """Gets build environment values
 
+    Args:
+        cfg (dict): dictionary holding full configuration (by default defined in app.cfg)
+
     Returns:
          dict(str, dict): dictionary of configuration data
     """
-    cfg = config.read_config()
+    fn = sys._getframe().f_code.co_name
+
+    # cfg = config.read_config()
     buildenv = cfg[BUILDENV]
 
     jobs_base_dir = buildenv.get(JOBS_BASE_DIR)
-    log("jobs_base_dir '%s'" % jobs_base_dir)
+    log(f"{fn}(): jobs_base_dir '{jobs_base_dir}'")
```

Comment:
Do we already have an open issue to enhance the `log` function so it automatically adds the function from which the log message was sent?

That would be way cleaner than spreading `{fn}()` all over the place in `log` calls, and shouldn't be very difficult I think.

No blocker here, can be cleaned up later since we're already doing this all over the place

Reply:
Hafsa was working on something like this (issue was https://github.com/EESSI/eessi-bot-software-layer/issues/110). However, it is not generally applicable. The logging definitely needs more thinking/rework.